cleanup.rb

```diff
@@ -46,12 +90,24 @@ def clean_up(client, repository, results_repository, results_path, age_limit)
   branch_history_limit = 20
   file_age_limit = 9000
 
+  if limit_reached
+    if results_path.end_with?(*long_running_branches)
+      branch_history_limit = long_running_branch_limit
+      logger.info("Long running branch limit reached: #{branch_history_limit}")
+    else
+      branch_history_limit = feature_branch_limit
+      logger.info("Feature branch limit reached: #{branch_history_limit}")
+    end
+  end
+
+  # These are the absolute failsafe limits required by github
+  # if we are approaching 1000 files in a single directory
   if files.size > 800
     if files.size > 999
       branch_history_limit = 1
       file_age_limit = age_limit + 1
     else
-      branch_history_limit = 5
+      branch_history_limit = [5, branch_history_limit].min
```

Comment:
So this overrides the limit to a maximum of 5 results, even for long running branches, if the files in the directory are approaching 1000?